openconcept/thermal/battery_cooling.py

```diff
@@ -0,0 +1,263 @@
+        self.add_input('T_in', shape=(nn,), units='K', val=300.)
+        self.add_input('T_battery', shape=(nn,), units='K', val=300.)
+        self.add_input('mdot_coolant', shape=(nn,), units='kg/s', val=0.20)
+        self.add_input('battery_weight', units='kg', val=478.)
+        self.add_input('n_cpb', units=None, val=82.)
+        self.add_input('t_channel', units='m', val=0.0005)
+
+        self.add_output('dTdt', shape=(nn,), units='K/s', tags=['integrate', 'state_name:T_battery', 'state_units:K', 'state_val:300.0', 'state_promotes:True'])
+        self.add_output('T_surface', shape=(nn,), units='K', lower=1e-10)
+        self.add_output('T_core', shape=(nn,), units='K', lower=1e-10)
+        self.add_output('q', shape=(nn,), units='W')
+        self.add_output('T_out', shape=(nn,), units='K', val=300, lower=1e-10)
+
+        self.declare_partials(['*'], ['*'], method='cs')
+
+    def compute(self, inputs, outputs):
+        nn = self.options['num_nodes']
+        n_cells = inputs['battery_weight'] * self.options['battery_weight_fraction'] / self.options['cell_mass']
+        n_bandoliers = n_cells / inputs['n_cpb'] / 2
+
+        mdot_b = inputs['mdot_coolant'] / n_bandoliers
+        q_cell = inputs['q_in'] / n_cells
+        hconv = self.options['nusselt'] * self.options['fluid_k'] / 2 / inputs['t_channel']
+
+        Hc = self.options['cell_height']
+        Dc = self.options['cell_diameter']
+        mc = self.options['cell_mass']
+        krc = self.options['cell_kr']
+        cpc = self.options['cell_specific_heat']
+        L_bandolier = inputs['n_cpb'] * Dc
+
+        cpf = self.options['coolant_specific_heat'] # of the coolant
+
+        A_heat_trans = Hc * L_bandolier * 2 # two sides of the tape
+        NTU = hconv * A_heat_trans / mdot_b / cpf
+        Kcell = mdot_b * cpf * (1 - np.exp(-NTU)) / 2 / inputs['n_cpb'] # divide out the total bandolier convection by 2 * n_cpb cells
+        # the convective heat transfer is (Ts - Tin) * Kcell
+        PI = np.pi
+        
+        Tbar = inputs['T_battery']
+        Rc = Dc / 2
+
+        K_cyl = 8*np.pi*Hc*krc
+
+        Ts = (K_cyl * Tbar + Kcell * inputs['T_in']) / (K_cyl + Kcell)
+        
+        outputs['T_surface'] = Ts
+
+        q_conv = (Ts - inputs['T_in']) * Kcell * n_cells
+        outputs['dTdt'] = (q_cell - (Ts - inputs['T_in']) * Kcell) / mc / cpc  # todo check that this quantity matches convection
+
+
+        outputs['q'] = q_conv
+
+        qcheck = (Tbar - Ts) * K_cyl
+        # UAcomb = 1/(1/hconv/A_heat_trans+1/K_cyl/2/inputs['n_cpb'])
+        # qcheck2 = (Tbar - inputs['T_in']) * mdot_b * cpf * (1 - np.exp(-UAcomb/mdot_b/cpf)) / 2 / inputs['n_cpb']
+
+        # if np.sum(np.abs(qcheck - outputs['q']/n_cells)) > 1e-5:
+        #     # the heat flux across the cell is not equal to the heat flux due to convection
+        #     raise ValueError('The surface temperature solution appears to be wrong')
```

Comment:
Is this commented out code needed for any reason?

Reply:
Not sure, but for now do you mind if we keep it in case a related issue comes up? I didn't add it--it's from Ben's original addition (it's already in OC source)

Reply:
Yeah if you feel it's easier to keep this around than it's fine with me.